examples/sierpinski.js

```diff
@@ -1,9 +1,10 @@
-// draw a Sierpinski Curve of arbitrary recursive depth
 // https://en.wikipedia.org/wiki/Sierpi%C5%84ski_curve
 
+/** helper for `sierpinski` */
 function halfSierpinski(size, level) {
-   if (level <= 0)
+   if (level <= 0) {
       forward(size);
+   }
    else {
       const part = () => {
```

Comment:
This on-the-fly function **part()** deserves to become a more formalized created-once function w/ 2 parameters:
```js
function part(size, level) {
   halfSierpinski(size, --level);
   left(45);
   forward(size * Math.SQRT2);
   left(45);
   halfSierpinski(size, level);
}
```
```js
      part(size, level);
      right(90);
      forward(size);
      right(90);
      part(size, level);
```

Reply:
Oh right, I didn't notice the duped code. Thanks! I'll do it in a minute...

Reply:
I was thinking, and got to this conclusion:
pros:
- less indent
- pure fn
- potentially better performance (I doubt it, because new engines are smart)

cons:
- more verbose, because of params, and explicit arg-passing

Reply:
> Potentially better performance (I doubt it, because new engines are smart).

I doubt there's anything an engine can do to create a function faster than invoking an already existing function; unless it keeps the temporary function around & re-use it.

> ... because there are 2 local-scoped parts (each inside different fns).

Well, in such case we can rename them so it is clear which function a helper function **part()** belongs to.

Reply:
I [renamed it](https://github.com/Rudxain/js-turtle/commit/ca15abfcdb9acfa654e3703d9c88bcee7868284e), then decided that it should [be a loop](https://github.com/Rudxain/js-turtle/commit/ad859a3a2cf51804e2d1cb4cc0f7fdae3ed2e855) lol

I used `_` as index, to make explicit that it's an "unused" var

Reply:
Although making it a loop is surely valid, aesthetically `for (let _=0; _<2; _++) {` is darn ugly!
Maybe making it an actual **half()** function is more visually pleasant.

Reply:
I agree. But what if we write it as?...
```js
for (const _ of range(2))
```

I've been considering adding a simple polyfill for the future `range` generator, but I need to verify if it's ES6-compatible. The `range` helper would be defined in turtle.js, making it available globally (I noticed it's needed in other places.

We could even add it to the public API! It would make JS more "Pythony" for users

Reply:
> , but I need to verify if it's ES6-compatible.

[`for...of`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of) and its [iterable protocol](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols) condition are all ES6-compatible.

> The **range()** helper would be defined in turtle.js, making it available globally...

That makes much sense and useful to simplify basic loops.

> `for (const _ of range(2))`

Visually much better now. Although I still think creating a loop merely to iterate twice seems off somehow.

Reply:
> [`for...of`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of) and its [iterable protocol](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols) condition are all ES6-compatible.

I meant [`function*`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/function*), but I searched it and it's also compatible. Nice!

> That makes much sense and useful to simplify basic loops.

Thanks

> creating a loop merely to iterate twice seems off somehow

I feel the same lol